api/client.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+import (
+	"github.com/google/uuid"
+	"sync"
+)
+
+type ClientRequestStore struct {
+	m sync.RWMutex
+	clientsR		map[string]*ClientRequest		//map with the client ip
+}
+
+func NewClientRequestStore() *ClientRequestStore {
+	crs := &ClientRequestStore{
+		clientsR: 	map[string]*ClientRequest{},
+	}
+	return crs
+}
+
+func (crs *ClientRequestStore) NewClientRequest(host string) *ClientRequest {
+
+	cl := &ClientRequest{
+		username:	  uuid.New().String(),
+		password:     uuid.New().String(),
+		cookies: 	  map[string]string{},
+		host:		  host,
+		ports:  	  map[string]uint{},
+		requestsMade: 0,
+	}
+
+	crs.clientsR[host] = cl
+	return cl
+}
+
+type ClientRequest struct {
+	username 		string
+	password 		string
+	cookies			map[string]string				//map cookie with challenges tag
```

Comment:
Theoretically speaking, only a single cookie should be required; each user gets a single cookie, then the state of this user is associated with the cookie. There is a natural distinction across events anyway I think, because (if I recall correctly) cookies apply to a full domain (i.e. `test.ntp-event.dk`), so cookies are not shared across events.

Then again, I don't know the exact implementation details of Haaukins, so this may require some refactoring to get it working right.